File: tasks_26/solutions/26-53.py
with open('26-53.txt') as F:
  N = int(F.readline())
  data = []
  for i in range(N):
     data.append( int(F.readline()) )
  #data = [int(s) for s in F]

data.sort()

averages = []
for i in range(N-1):
  for j in range(i+1,N):
    if data[i] % 2 == 1 and data[j] % 2 == 1:
      s = data[i] + data[j]
      averages.append( s//2 )
averages.sort()


selected = []
i = 0
for av in averages:
  while i < N and data[i] < av:
    i += 1
  if i < N and data[i] == av:
    selected.append(av)

print( len(selected), selected[-1] )

File: tasks_26/solutions/26-53.pas
##
var F:= openRead('26.txt');
var n:= F.ReadInteger;
var data:= (1..n).Select(i->F.ReadInteger)
           .Sorted.ToArray;

var averages := new List<integer>;
for var i:=0 to data.High-1 do
  for var j:=i+1 to data.High do
    if (data[i] mod 2 = 1) and (data[j] mod 2 = 1) then 
      averages.Add( (data[i]+data[j]) div 2 );
averages.Sort;

var selected := new List<integer>;
var i:= 0;
foreach var av in averages do begin
  while (i < N) and (data[i] < av) do
    i += 1;
  if (i < N) and (data[i] = av) then
    selected.Add(av);
end;

print( selected.Count, selected.Last )


